src/frequenz/sdk/data_ingestion/resampling/timeseries_resampling_actor.py

```diff
@@ -0,0 +1,159 @@
+"""
+TimeSeriesResamplerActor used to subscribe for resampled component metrics.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+import logging
+from typing import Callable, Dict, Optional, Sequence
+
+from frequenz.channels import Broadcast, MergeNamed, Receiver, Select, Timer
+
+from frequenz.sdk.actor import actor
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+from frequenz.sdk.data_ingestion.resampling.data_provider import DataProvider
+from frequenz.sdk.data_ingestion.resampling.subscriptions import (
+    Subscription,
+    SubscriptionRequest,
+    SubscriptionResponse,
+)
+from frequenz.sdk.data_ingestion.resampling.timeseries_resampler import (
+    TimeSeriesResampler,
+)
+
+logger = logging.Logger(__name__)
+
+ResamplingFunction = Callable[[Sequence[Sample], float], float]
+
+
+# pylint: disable=unused-argument
+def average(samples: Sequence[Sample], resampling_frequency: float) -> float:
+    """Calculate average of the provided values.
+
+    Args:
+        samples: sequences of samples to apply the average to
+        resampling_frequency: resampling frequency, in seconds
+
+    Returns:
+        average of all the sample values
+    """
+    if len(samples) == 0:
+        return 0.0
+    return sum(sample.value for sample in samples) / len(samples)
+
+
+@actor
+class TimeSeriesResamplingActor:
+    """TimeSeriesResamplingActor used to ingest component data and resample it."""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        data_provider: DataProvider,
```

Comment:
This will probably go away.

Reply:
Yes, this is meant to go away once I know how `DataSource`'s interface looks like.